src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -256,12 +256,41 @@
     If a user requests that any @(AdditionalFiles) items are copied to the output directory
     we add them to the @(None) group to ensure they will be copied.
  -->
- 
-   <Target Name="CopyAdditionalFiles" 
-           BeforeTargets="AssignTargetPaths">
+
+  <Target Name="CopyAdditionalFiles"
+          BeforeTargets="AssignTargetPaths">
     <ItemGroup>
       <None Include="@(AdditionalFiles)" Condition="'%(AdditionalFiles.CopyToOutputDirectory)' != ''" />
     </ItemGroup>
   </Target>
 
+  <!--
+    ========================
+    GeneratedFilesOutputPath
+    ========================
+    
+    Controls output of generated files.
+    
+    GeneratedFilesOutputPath controls the location the files will be output to.
+    The compiler will not emit any generated files when the path is empty, and defaults to a /generated directory in $(IntermediateOutputPath) if $(IntermediateOutputPath) has a value.
+    EmitGeneratedFiles allows the user to control if anything is emitted by clearing the property when not true.
+    When GeneratedFilesOutputPath is set we ensure the directory exists, and issue a warning if EmitGeneratedFiles is true, but no GeneratedFilesOutputPath is set
+    -->
+  <PropertyGroup>
+    <EmitGeneratedFiles Condition="'$(EmitGeneratedFiles)' == ''">true</EmitGeneratedFiles>
```

Comment:
A tad worried this name may be too general. Can't think of a great way to check though as it's hard to search all props / targets on GitHub. 

My suggestion would be to change the names to: `EmitCompilerGeneratedFiles` and `CompilerGeneratedFilesOutputPath`